### DISCUSSION TITLE
Coverage Report

### DISCUSSION THREAD

**Discussion by joshuafolkken** [MikeSchulze/gdUnit4#568] [Q&A] [answered]
Hi, 

How can I check the Coverage Report?

**Comment by MikeSchulze** [comment, marked as answer]
There is no coverage report available for GdUnit4. Godot do not provide any tools to build it.

Only for GdUnit4Net coverage reports is available by the c# tools.

**Reply by joshuafolkken** [reply]
Thank you for your answer.

**Comment by Bradley-Labs** [comment]
I am confused, I am using Godot .net and rider and get no display within rider when I run the tests

**Reply by MikeSchulze** [reply]
Can you please give more details?
If your issue related to the Code Coverage? If not, please do not reuse this conversation.

**Comment by CodeSmith32** [comment]
Hey @MikeSchulze,

> Godot do not provide any tools to build it.

Believe it or not, this actually isn't entirely true!

It looks like [someone accomplished code coverage](https://github.com/jamie-pate/godot-code-coverage) for the GUT unit testing library. It's not trivial, but Godot makes the [`source_code` property on the Script class](https://docs.godotengine.org/en/stable/classes/class_script.html#class-script-property-source-code) writable. After updating `source_code` and calling `reload()`, apparently the script executes the new code. This is what [that GitHub repo uses under the hood](https://github.com/jamie-pate/godot-code-coverage/blob/main/addons/coverage/coverage.gd#L260).

Maybe worth looking into? Not pushing. Just thought it was really interesting!

**Reply by MikeSchulze** [reply]
> Hey @MikeSchulze,
> 
> > Godot do not provide any tools to build it.
> 
> Believe it or not, this actually isn't entirely true!

The tool was outdated at this time and only supported Godot 3.5.

I build my own coveage tool with no source code modification is need, it has nearly no performance impact.
The [open beta](https://github.com/godot-gdunit-labs/gdUnit4-coverage) has already startet.

<img width="2271" height="1365" alt="image" src="https://github.com/user-attachments/assets/234b6bcc-87f5-4d39-9b98-7f3846b00e02" />
